### PR TITLE
Upgrading to newest metrics version

### PR DESCRIPTION
**Pull request by alobodzki**
Moving to a newest 3.2.6 dropwizard metrics.

### CODE REVIEW

File: components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteReporter.java [hunk 1]
```diff
@@ -230,12 +226,20 @@ public void report(SortedMap<String, Gauge> gauges,
...
 
     private <M extends Metric> void doReport(String name, M metric, long timestamp, MetricReportingAction<M> consumer) {
         try {
+            if (!graphite.isConnected()) {
+                graphite.connect();
+            }
+
             consumer.execute(name, metric, timestamp);
         } catch (Exception e) {
             LOGGER.error("Error reporting metric '" + name + "': " + e.getMessage(), e);
```
Comment (mikkokar) [inline components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteReporter.java]: Hi @alobodzki could you provide some background for this change? Mainly for the record please.

File: components/proxy/src/test/java/com/hotels/styx/admin/handlers/MetricsHandlerTest.java [hunk 2]
```diff
@@ -44,6 +44,6 @@ public void respondsToRequestWithJsonResponse() {
     public void exposesRegisteredMetrics() {
         metricRegistry.counter("foo").inc();
         FullHttpResponse response = waitForResponse(handler.handle(get("/metrics").build()));
-        assertThat(response.bodyAs(UTF_8), is("{\"version\":\"3.0.0\",\"gauges\":{},\"counters\":{\"foo\":{\"count\":1}},\"histograms\":{},\"meters\":{},\"timers\":{}}"));
+        assertThat(response.bodyAs(UTF_8), is("{\"version\":\"3.1.3\",\"gauges\":{},\"counters\":{\"foo\":{\"count\":1}},\"histograms\":{},\"meters\":{},\"timers\":{}}"));
```
Comment (mikkokar) [inline components/proxy/src/test/java/com/hotels/styx/admin/handlers/MetricsHandlerTest.java]: Just curious as why it is reporting version 3.1.3 when we are upgrading to 3.2.6?

File: components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteReporter.java (hunk 1 above)
Comment (alobodzki) [reply components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteReporter.java]: Our implementation of GraphiteReport tries to reconnect between failures in a single upload, but the implementation of reconnectIfNecessary reconnects only with IOException, there is no default option in that method, so theoretically we could enter doReport method without connected GraphiteConsole. Dropwizard implementation doesn't have this issue, as they don't try to recover, they just fail fast.

File: components/proxy/src/test/java/com/hotels/styx/admin/handlers/MetricsHandlerTest.java (hunk 2 above)
Comment (alobodzki) [reply components/proxy/src/test/java/com/hotels/styx/admin/handlers/MetricsHandlerTest.java]: I assumed it's JSON scheme version: com.codahale.metrics.json.MetricsModule#VERSION

Comment (kvosper) [reply components/proxy/src/test/java/com/hotels/styx/admin/handlers/MetricsHandlerTest.java]: It's definitely not the library version as that one was 3.1.1 before, but the JSON version was 3.0.0

**Pull request merged**